install.sh

```diff
@@ -59,19 +57,21 @@ cleanup() {
 }
 
 trap cleanup EXIT
+
+echo Downloading the archive...
+
 HTTP_CODE=$(curl -SL --progress-bar "$DOWNLOAD_URL" --output "$TEMP_FILE" --write-out "%{http_code}")
-if [[ ${HTTP_CODE} -lt 200 || ${HTTP_CODE} -gt 299 ]]; then
+if [ "$HTTP_CODE" -lt 200 ] || [ "$HTTP_CODE" -gt 299 ]; then
   echo "error: your platform and architecture (${PLATFORM}-${ARCH}) is unsupported."
   exit 1
 fi
 
-tar xzf "$TEMP_FILE" -C "$TEMP_FOLDER"
+tar xzf "$TEMP_FILE" -C "$TEMP_FOLDER" codecrafters
 
-chmod 0755 "$TEMP_FOLDER/codecrafters"
-if ! mv "$TEMP_FOLDER/codecrafters" "$INSTALL_PATH" 2> /dev/null; then
-  sudo -k mv "$TEMP_FOLDER/codecrafters" "$INSTALL_PATH"
+if ! install "$TEMP_FOLDER/codecrafters" "$INSTALL_PATH" 2>/dev/null; then
```

Comment:
@nikandfor how sure are we that `install` is a good option re: compatibility? It looks like popular scripts don't use `install` and use `chmod`/`mv` instead, example:

- https://sentry.io/get-cli/
- https://fly.io/install.sh
- https://deno.land/x/install/install.sh

Reply:
(I saw your note in the description re: `+x` being disabled in tmp, just want to confirm whether this is best/common practice)

Reply:
Agree, this is probably not much compatible. Even flags are totally different. Maybe change the order of mv+chmod then?

Reply:
@nikandfor Yep, that works!

Reply:
What is the purpose of -k option for sudo here?

Reply:
I just copied the `sudo -k` from one of the scripts linked to above - honestly had no clue what it does 🙂 I think it's fine to `chmod` in the tmp folder as you suggested. I'll keep an eye out for user complaints. Fwiw I haven't heard any complaints re: `chmod` failing yet, but we have heard a couple about posix sh compatibility already.

In my opinion, If there isn't an "obvious" answer to something, averaging approaches from the scripts linked above is a good strategy - these are used by 1000s of people, so there's some quality control already baked in.